examples/computer_vision/vision_transformer/det_callback.py

```diff
@@ -0,0 +1,187 @@
+
+            self.last_metrics.update(metrics)
+
+    def get_metrics(self, logs: typing.Dict) -> typing.Tuple[typing.Dict, str]:
+        metrics = logs
+        metric_type = get_metric_type(logs)
+        if self.filter_metrics:
+            metrics = {}
+            for k, v in logs.items():
+                if any(m in k for m in self.filter_metrics) is True:
+                    metrics[k] = v
+
+        return metrics, metric_type
+
+    def on_save(self, args: TrainingArguments, state: TrainerState, control: TrainerControl, **kwargs):
+        info = self._det.get_cluster_info()
+        if info is None:
+            # TODO: modify to support local mode
+            logging.warning('ClusterInfo is None: not running in a task. Skip saving.')
+            return
+
+        if state.is_world_process_zero:
+            save_path = os.path.join(args.output_dir, f"checkpoint-{state.global_step}")
+            det_checkpoint_metadata = {"steps_completed": state.global_step, "trial_id": info.trial.trial_id}
+
+            if self.tokenizer_options is not None:
+                det_checkpoint_metadata['tokenizer_options'] = self.tokenizer_options
+
+            if self.tokenizer is not None:
+                self.tokenizer.save_pretrained(os.path.join(save_path, "tokenizer"))
+
+            if self.user_checkpoint_metadata is not None:
+                self._on_save_user_data(save_path)
+
+            self.core_context.checkpoint.upload(save_path, det_checkpoint_metadata)
+
+        if self.core_context.preempt.should_preempt():
+            raise Exception("Process preempted / killed")
+
+    def _on_save_user_data(self, save_path) -> None:
+        '''
+        User-defined saving of objects from self.checkpoint_metadata under save_path.
+        After objects are saved, Determined handles uploading and downloading objects to/from selected storage.
+        '''
+        raise NotImplementedError("No implementation for _on_save_user_data. "
+                                  "Objects passed to the callback via checkpoint_metadata will not be saved.")
+
+    def load_last_checkpoint(self, args: typing.Dict) -> None:
+        info = self._det.get_cluster_info()
+
+        if info is None:
+            # TODO: modify to support local mode
+            logging.warning('ClusterInfo is None: not running in a task. Skip loading.')
+            return
+
+        latest_checkpoint = info.latest_checkpoint
+        if latest_checkpoint is not None:
+            metadata = self.core_context.checkpoint.get_metadata(latest_checkpoint)
+            prev_trial_id = metadata["trial_id"]
+            trial_id = info.trial.trial_id
+            if trial_id != prev_trial_id:
+                resume_step = 0
+            else:
+                resume_step = metadata['steps_completed']
+            checkpoint_path = os.path.join(args.output_dir, f"checkpoint-{resume_step}")
+            self.core_context.checkpoint.download(latest_checkpoint, checkpoint_path)
```

Comment:
would all ranks perform this download? we should probably limit it to local rank 0 processes on each node so we do not duplicate multiple times per node